src/app/services/configuration.services.ts

```diff
@@ -0,0 +1,56 @@
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+
+import * as fromApp from '../store';
+import {Store, select} from '@ngrx/store';
+import {HttpClient, HttpErrorResponse} from '@angular/common/http';
+import {ConfigurationModel} from '../models/configuration.model';
+import {catchError, take} from 'rxjs/operators';
+import {throwError} from 'rxjs';
+
+
+/**
+ *  Configuration Services responsible for fetching initial cofig data needed for app  to run
+ *  It also provides instance of environment or other data;
+ */
+@Injectable()
+export class AppConfigService {
+
+  constructor(private http: HttpClient,  private store: Store<fromApp.State>) {}
+  private configuration: ConfigurationModel;
+
+  load(): Observable<any> {
+    const jsonFile = `assets/config/config.json`;
+    return this.http.get(jsonFile).pipe(
+      catchError(this.handleError)
+    );
+  }
+
+  setConfiguration() {
+    this.store.pipe(select(fromApp.getAppFeatures), take(1)).subscribe(config => {
+      this.configuration = config;
+    });
+  }
+  getFeatureToggle() {
+    return this.configuration.features;
+  }
+
+  // todo make it global and make it make sense
```

Comment:
I think it is best to stick the // todo in Jira so you can keep track

Reply:
👍